vllm/core/block/interfaces.py

```diff
@@ -99,18 +106,21 @@ def content_hash(self) -> Optional[int]:
 class BlockAllocator(ABC):
 
     @abstractmethod
-    def allocate_mutable_block(self, prev_block: Optional[Block]) -> Block:
+    def allocate_mutable_block(self, prev_block: Optional[Block],
+                               contextual_hash: Optional[int]) -> Block:
```

Comment:
Ditto to all similar places.
```suggestion
                               contextual_hash: Optional[int] = None) -> Block:
```